### PR TITLE
OE-core master update + Yocto Compatible 2.0 enhancements

### PR DESCRIPTION
**Pull request by pohly**
Imports and adapts to latest OE-core master, plus uses all the Yocto Compatible 2.0 support that recently landed in OE-core + bitbake master:

* optional includes
* oe.utils.all/any_distro_features()
* OE-core distrooverrides.bbclass
* BBFILES_DYNAMIC for .bbappends which may or may not have a .bb

Also split up the Yocto Compatible 2.0 selftest, to make it easier to run just the signature checks.

### CODE REVIEW

**Comment by pohly** [comment]
test this please

I had not caught all _refkit -> _df-refkit renames.

**Comment by pohly** [comment]
test this please

**Comment by pohly** [comment]
retest this please

Seems like an unrelated problem (race?):
https://ostroproject.org/jenkins/job/intel-iot-refkit_pull-requests/1410/consoleFull
```
DEBUG: SITE files ['endian-little', 'common-linux', 'common-glibc', 'bit-64', 'x86_64-linux', 'common']
00:37:57.075 | DEBUG: Executing shell function do_install_ptest_base
00:37:57.075 | cp: cannot stat '/srv/jenkins/workspace/ci-2017-06-13_12-39-06-build-1410/build/tmp-glibc/work/corei7-64-refkit-linux/util-linux/2.30-r0/util-linux-2.30/tests/ts/tailf': No such file or directory
00:37:57.075 | WARNING: /srv/jenkins/workspace/ci-2017-06-13_12-39-06-build-1410/build/tmp-glibc/work/corei7-64-refkit-linux/util-linux/2.30-r0/temp/run.do_install_ptest_base.160077:1 exit 1 from 'cp -pR /srv/jenkins/workspace/ci-2017-06-13_12-39-06-build-1410/build/tmp-glibc/work/corei7-64-refkit-linux/util-linux/2.30-r0/util-linux-2.30/tests/ts/$d /srv/jenkins/workspace/ci-2017-06-13_12-39-06-build-1410/build/tmp-glibc/work/corei7-64-refkit-linux/util-linux/2.30-r0/image/usr/lib/util-linux/ptest/tests/ts/'
00:37:57.075 | ERROR: Function failed: do_install_ptest_base (log file is located at /srv/jenkins/workspace/ci-2017-06-13_12-39-06-build-1410/build/tmp-glibc/work/corei7-64-refkit-linux/util-linux/2.30-r0/temp/log.do_install_ptest_base.160077)
00:37:57.075 NOTE: recipe util-linux-2.30-r0: task do_install_ptest_base: Failed
```

**Comment by pohly** [comment]
The util-linux 2.30 upgrade from OE-core master-next was broken. Trying again without it.

retest this please

test this please

**Comment by pohly** [comment]
retest this please

test this please

**Comment by pohly** [comment]
OE-core test branch hadn't been pushed.

retest this please

**Comment by pohly** [comment]
test this please

**Comment by pohly** [comment]
Rebased onto current master, fixed ovmf (in OE-core), fixed upm (in refkit).

**Comment by pohly** [comment]
@mythi: do we really want packagegroup-sensors in refkit-image-common? It drags in mraa, upm, nodejs, and the latter is what currently fails in the "test_common_poky_config" test, because it builds refkit-image-common with the Poky distro config and thus gcc7, which is known to fail for nodejs.

**Comment by mythi** [comment]
On Fri, Jun 16, 2017 at 11:32 PM, Patrick Ohly wrote:

> @mythi <https://github.com/mythi>: do we really want packagegroup-sensors
> in refkit-image-common? It drags in mraa, upm, nodejs, and the latter is
> what currently fails in the "test_common_poky_config" test, because it
> builds refkit-image-common with the Poky distro config and thus gcc7, which
> is known to fail for nodejs.
>
>
>
I'm pretty sure I dropped some time ago (moved it to gateway only). If
that's not the case, please remove.

**Comment by mythi** [comment]
On Sat, Jun 17, 2017 at 8:02 AM, Ylinen, Mikko <mikko.ylinen@intel.com>
wrote:

>
>
> On Fri, Jun 16, 2017 at 11:32 PM, Patrick Ohly wrote:
>
>> @mythi <https://github.com/mythi>: do we really want
>> packagegroup-sensors in refkit-image-common? It drags in mraa, upm, nodejs,
>> and the latter is what currently fails in the "test_common_poky_config"
>> test, because it builds refkit-image-common with the Poky distro config and
>> thus gcc7, which is known to fail for nodejs.
>>
>>
>>
> I'm pretty sure I dropped some time ago (moved it to gateway only). If
> that's not the case, please remove.
>
>
Ah, it was iotivity. Let's move sensors in gateway.

**Comment by pohly** [comment]
Rebased (again).

Instead of reducing the content of meta-refkit-common, I restricted the test_common_poky_config to OE-core + meta-refkit-core. That's the part that should work. Additional layers will only work when using our distro config, because that's where we enabled workarounds for known problems.

Reducing meta-refkit-common can and should still be done separately.

The upm + gcc7 incompatibility gets  resolved via backporting a fix (problem still occurs while sensors are in meta-refkit-common). Someone more familiar with upm should do a major version upgrade.

**Comment by pohly** [comment]
@ipuustin can you please review and merge, if it passes?

File: meta-refkit-core/bbappends/meta-security/recipes-tpm/swtpm/swtpm-wrappers-native.bbappend [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Temporary workaround (needs to be fixed in meta-security once the
+# necessary patch "net-tools: enable native and nativesdk variant"
+# is in OE-core): swtpm_setup.sh needs netstat command.
```
Comment (ipuustin) [inline meta-refkit-core/bbappends/meta-security/recipes-tpm/swtpm/swtpm-wrappers-native.bbappend]: Appears to be in oe-core (https://github.com/openembedded/openembedded-core/commit/508163bef169cf0d9aa97e73c02d1ecc68480e91). Can this be removed?

**Comment by pohly** [comment]
On Mon, 2017-06-19 at 01:22 -0700, Ismo Puustinen wrote:
> @ipuustin commented on this pull request.
> 
> 
> ______________________________________________________________________
> 
> In
> meta-refkit-core/bbappends/meta-security/recipes-tpm/swtpm/swtpm-wrappers-native.bbappend:
> 
> > @@ -0,0 +1,4 @@
> +# Temporary workaround (needs to be fixed in meta-security once the
> +# necessary patch "net-tools: enable native and nativesdk variant"
> +# is in OE-core): swtpm_setup.sh needs netstat command.
> 
> Appears to be in oe-core (openembedded/openembedded-core@508163b). Can
> this be removed?

meta-security also needs to be fixed first. I haven't gotten around to
sending a patch for that.

File: meta-refkit/lib/oeqa/selftest/cases/refkit_poky.py [hunk 2]
```diff
@@ -31,6 +31,24 @@
...
 from oeqa.utils.commands import runCmd, bitbake, get_bb_var, get_bb_vars, runqemu
 
+def find_layers():
+    layers = {}
+    result = runCmd('bitbake-layers show-layers')
+    for layer, path, pri in re.findall(r'^(\S+) +([^\n]*?) +(\d+)$', result.output, re.MULTILINE):
+        layers[layer] = path
+    # meta-poky should not be active. We expect it next to the meta-refkit layer.
+    if not 'meta-poky' in layers:
+        layers['meta-poky'] = os.path.join(os.path.dirname(layers['meta-refkit']), 'meta-yocto', 'meta-poky')
```
Comment (ipuustin) [inline meta-refkit/lib/oeqa/selftest/cases/refkit_poky.py]: Should you check if the layer was actually there?

**Comment by pohly** [comment]
On Mon, 2017-06-19 at 02:30 -0700, Ismo Puustinen wrote:
> @ipuustin commented on this pull request.
> 
> 
> ______________________________________________________________________
> 
> In meta-refkit/lib/oeqa/selftest/cases/refkit_poky.py:
> 
> > @@ -31,6 +31,24 @@
>  from oeqa.selftest.case import OESelftestTestCase
>  from oeqa.utils.commands import runCmd, bitbake, get_bb_var, get_bb_vars, runqemu
>  
> +def find_layers():
> +    layers = {}
> +    result = runCmd('bitbake-layers show-layers')
> +    for layer, path, pri in re.findall(r'^(\S+) +([^\n]*?) +(\d+)$', result.output, re.MULTILINE):
> +        layers[layer] = path
> +    # meta-poky should not be active. We expect it next to the meta-refkit layer.
> +    if not 'meta-poky' in layers:
> +        layers['meta-poky'] = os.path.join(os.path.dirname(layers['meta-refkit']), 'meta-yocto', 'meta-poky')
> 
> Should you check if the layer was actually there?

I don't think it is worth checking. We know it's there because we put it
there, and (re)moving it will cause test failures.

**Pull request merged**